### PR TITLE
Added command line support for OPC-UA method calls according to #321

### PR DESCRIPTION
**Pull request by SCHUNK-Osswald**
Finished the command line tool for calling OPC_UA methods as asked for in #321.

Method calling now works on PC (windows 7) and BBB (Debian Linux on ARM) with our proprietary OPC-UA server.

Not sure if this whole git-fork-clone-branch-commit-push stuff worked as expected. (Maybe my first changes ended in master rather than in the add_call_cli branch.)

Feedback / suggestions / corrections welcome.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -7,8 +7,8 @@
 else:
     install_requires = ["python-dateutil"]
 
-setup(name="freeopcua", 
-      version="0.10.17",
+setup(name="freeopcua",
+      version="0.10.17.1",
```
Comment (oroulet) [inline setup.py]: no need to increase version here.

**Comment by oroulet** [comment]
thanks for the PR. it looks OK. Btw, may I ask where does SCHUNK has an opc ua server? I have a bunch of schunck hardware, but they tend to use older protocol/connections like serial ;-)

File: setup.py (hunk 1 above)
Comment (SCHUNK-Osswald) [inline setup.py]: OK, sorry. Just did that so that pip performs the update...

**Comment by SCHUNK-Osswald** [comment]
Yes, we tend to use older HW, but even for us serial (RS232) is pretty uncommon nowadays. Most of our modules use CAN or PROFIbus. Newer producst use PROFInet, Ethernet or USB.

And now OPC-UA is used within a research project that we participate in. Unfortunately I guess it will be some time until this reaches buyable SCHUNK products...

BTW: Which products do you use?

File: opcua/common/node.py [hunk 2]
```diff
@@ -525,30 +525,30 @@ def delete(self, delete_references=True):
...
     def add_method(self, *args):
         return opcua.common.manage_nodes.create_method(self, *args)
- 
+
     def add_reference_type(self, parent, nodeid, bname):
         return opcua.common.manage_nodes.create_reference_type(parent, nodeid, bname)
-    
-    def call_method(parent, methodid, *args):
+
+    def call_method(self, parent, methodid, *args):
```
Comment (oroulet) [inline opcua/common/node.py]: This was not an error. Self was called parent her. I admit this was confusing. But you need to remind listener arg if you add self. Currently your pr breaks tests

File: opcua/tools.py [hunk 3]
```diff
@@ -651,3 +651,77 @@ def uahistoryread():
...
+                    break
+
+        if ( method_id is None):
+            # last resort:
+            method_id = ua.NodeId( identifier=args.method )#, namespaceidx=? )#, nodeidtype=?): )
+
+        #print "method_id=%s\nval=%s" % (method_id,val)
+
+        result_variants = node.call_method( parent, method_id, *val )
+        print "resulting result_variants=%s" % result_variants
```
Comment (oroulet) [inline opcua/tools.py]: No need for parent arg. The node itself is the parent

File: opcua/common/node.py (hunk 2 above)
Comment (SCHUNK-Osswald) [inline opcua/common/node.py]: I don't know what you mean by 

> But you need to remind listener arg if you add self.

?
If no aditional _parent_ is needed then lets just rename _parent_ to _self_ (in your original version) or remove _parent_ in my version.
But I think naming the first parameter of a member function other than _self_ will always cause trouble.

File: opcua/tools.py (hunk 3 above)
Comment (SCHUNK-Osswald) [inline opcua/tools.py]: Yes ok, would be clear if _parent_ parameter was called _self_, as said above.

BTW: What am I supposed to do now (regarding git/pull/push...)? Should I change my add_call_cli branch, then commit, then push, then create pull request? Or what?

**Comment by oroulet** [comment]
Looks like something went wrong with keyboard auto correction. The error
was that the first argument was called parent. Rename it to self, and
remove extra parent argument

**Comment by oroulet** [comment]
@SCHUNK-Osswald Can you finish this?

**Comment by SCHUNK-Osswald** [comment]
Am I done now or am I missing something?
(Still confused by this git workflow...)

**Comment by oroulet** [comment]
Looks good now. But you have one fail left: Then I can merge
SyntaxError: Missing parentheses in call to 'print'

ERROR

test_uawrite (tests_cmd_lines.TestCmdLines) ... Traceback (most recent call last):

  File "tools/uawrite", line 7, in <module>

    from opcua.tools import uawrite

  File "/home/travis/build/FreeOpcUa/python-opcua/tools/../opcua/tools.py", line 721

    print "resulting result_variants=%s" % result_variants

                                       ^

SyntaxError: Missing parentheses in call to 'print'

**Comment by oroulet** [comment]
Thanks for you contribution!

**Pull request merged**